src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -1,42 +1,62 @@
 import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
-import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
 import { OrderActionCreator, OrderRequestBody } from '../../../order';
 import { OrderFinalizationNotRequiredError } from '../../../order/errors';
 import { PaymentArgumentInvalidError } from '../../errors';
 import PaymentActionCreator from '../../payment-action-creator';
 import { PaymentRequestOptions } from '../../payment-request-options';
 import PaymentStrategy from '../payment-strategy';
 
+import { PaypalCommercePaymentProcessor, PaypalCommerceRequestSender } from './index';
+
 export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
 
     constructor(
         private _store: CheckoutStore,
         private _orderActionCreator: OrderActionCreator,
-        private _paymentActionCreator: PaymentActionCreator
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender,
+        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
     ) {}
 
-    initialize(): Promise<InternalCheckoutSelectors> {
+    async initialize(): Promise<InternalCheckoutSelectors> {
         return Promise.resolve(this._store.getState());
     }
 
     async execute(payload: OrderRequestBody, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         const { payment, ...order } = payload;
-        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId);
 
         if (!payment) {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
-        if (!paymentMethod || !paymentMethod.initializationData.orderId) {
-            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        if (!paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        let { orderId } = paymentMethod.initializationData;
+
+        if (!orderId) {
```

Comment:
I assume when you call `setupPayment`, the server will go and talk to PayPal, get an order ID from them, and set it in the session. That order ID will then be surfaced through `initializationData`. So what happens if the shopper doesn't complete whatever they have to do in the PayPal modal? It seems when this method is called again, because the order ID is already defined, we won't be showing the modal again to the shopper?

Reply:
In the modal user approves payment in PayPal. If  it doesn't complete, paymentPayPal throw error and user sees error in the standard modal window
`setupPayment` should be called immediately before call paymentPayPal for actual data